src/App.tsx

```diff
@@ -15,36 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_BY_ALPHABET = 'alphabet',
+  SORT_BY_LENGTH = 'length',
+}
+
+interface Goods {
+  sortField: SortType | '';
+  reverse: boolean;
+}
+
+function getPreparedGoods(goods: string[], { sortField, reverse }: Goods) {
+  const preparedGoods = [...goods];
+
+  switch (sortField) {
+    case SortType.SORT_BY_ALPHABET:
+      preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+    case SortType.SORT_BY_LENGTH:
+      preparedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [reverse, setReverse] = useState(false);
```

Comment:
```suggestion
  const [isReversed, setIsReverse] = useState(false);
```
it's good practice for boolean variable names: https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig